### PR TITLE
chore: test cleanups

### PR DESCRIPTION
**Pull request by rwoll**
A small follow up MR for my last contributions.

### CODE REVIEW

File: tests/async/test_dialog.py [hunk 1]
```diff
@@ -22,64 +21,69 @@
...
-    async def on_dialog(dialog: Dialog):
+    def on_dialog(dialog: Dialog):
+        result.append(True)
         assert dialog.type == "prompt"
         assert dialog.defaultValue == "yes."
         assert dialog.message == "question?"
-        on_dialog_actions_complete.set_result(await dialog.accept("answer!"))
+        asyncio.create_task(dialog.accept("answer!"))
 
     page.on("dialog", lambda dialog: asyncio.create_task(on_dialog(dialog)))
```
Comment (mxschmitt) [inline tests/async/test_dialog.py:48]: You are creating two asyncio tasks once the dialog event will be received. Only one would be needed. We try to only wrap as less functions inside a task. So I would do it like shown here.
```suggestion
    page.on("dialog", on_dialog)
```

File: tests/async/test_dialog.py [hunk 2]
```diff
@@ -22,64 +21,69 @@
...
-    on_dialog_actions_complete: Future[None] = asyncio.Future()
+    result = []
 
-    async def on_dialog(dialog: Dialog):
-        on_dialog_actions_complete.set_result(await dialog.dismiss())
+    def on_dialog(dialog: Dialog):
+        result.append(True)
+        asyncio.create_task(dialog.dismiss())
 
     page.on("dialog", lambda dialog: asyncio.create_task(on_dialog(dialog)))
```
Comment (mxschmitt) [inline tests/async/test_dialog.py:60]: Would be also two tasks and a single lambda would clean it up.

File: tests/async/test_dialog.py [hunk 3]
```diff
@@ -22,64 +21,69 @@
...
```
[the remark this answers is not in the document]
Comment (rwoll) [reply tests/async/test_dialog.py]: Fixed in latest commit throughout. Thank you!

File: tests/async/test_dialog.py (hunk 1 above)
Comment (rwoll) [reply tests/async/test_dialog.py]: 🤦 Thank you! That had meant to go with these cleanups! Fixed in latest commit—at least it should be :)

**Comment by rwoll** [comment]
@mxschmitt Thanks for the feedback! Feel free to take another quick pass!

**Comment by rwoll** [comment]
With the latest `lambda` rewrite I'm running into `mypy` linter:

```
tests/async/test_dialog.py:57: error: "append" of "list" does not return a value
tests/async/test_dialog.py:67: error: "append" of "list" does not return a value
tests/async/test_dialog.py:77: error: "append" of "list" does not return a value
```

**Comment by rwoll** [comment]
@mxschmitt Whats your mitigation suggestion for the mypy error? The non-lambda code did not have that issue since it was a full function, could be multiline, didn't have to return everything etc.

**Comment by mxschmitt** [comment]
> @mxschmitt Whats your mitigation suggestion for the mypy error? The non-lambda code did not have that issue since it was a full function, could be multiline, didn't have to return everything etc.

Then go with the multi line approach, should be fine for now. In the meantime I try to understand the mypy error.

**Pull request merged**